raiden/utils/__init__.py

```diff
@@ -341,3 +341,18 @@ def split_in_pairs(arg: Iterable) -> Iterable[Tuple]:
     # from the iterator each time and produces the desired result.
     iterator = iter(arg)
     return zip_longest(iterator, iterator)
+
+
+class releasing:
```

Comment:
yep, but that is intentional, I know it doesn't follow our usual naming style, but this needs to be a class because of the object protocol, and I tried to make it look like `closing`

Reply:
Why does it require to be a class? Which protocol?

Closing [uses a decorator](https://docs.python.org/3.6/library/contextlib.html#contextlib.closing) from what I see.

Reply:
https://docs.python.org/3.6/reference/datamodel.html#with-statement-context-managers

Reply:
https://github.com/python/cpython/blob/3.6/Lib/contextlib.py#L163

Reply:
> Would using the @contextmanager decorator and a generator not yield the same thing?

Yes, it should be the same thing as the docs says.

If you want I can change it to `CamelCase`, no biggie.